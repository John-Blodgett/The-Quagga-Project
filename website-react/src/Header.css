#web-header-master {
    width: 100vw;
    height: 10vh;
    background-color: #0D3B66;
}

#web-header-logo {
    display: inline-block;
    
    position: inherit;
}

#web-header-title {
    display: inline-block;
    font-size: xx-large;
    font-weight: bold;
    color: white;
}

#zebra-background {
    display: table-cell;
    width: inherit;
    height: inherit;
    vertical-align: middle;
    background-image: url('./images/Zebra\ 5.png');
}

#web-cog-wheel {
    float: right;
    padding-right: 1.75vw;
    padding-top: 2vh;
}

#web-cog-wheel > img {
    width: 30px;
    height: 30px;
}