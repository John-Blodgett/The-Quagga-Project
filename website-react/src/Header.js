import React from 'react';
import './Header.css';
import logo from './images/Logo.png'

function Header() {
  return (
    <>
        <div id="web-header-master">
            <div id="zebra-background">
                <div id="web-header-logo">
                    <img src={logo} alt="Quagga Logo"></img>
                </div>
                <div id="web-header-title">
                    Quagga
                </div>
                <div id="web-cog-wheel">
                    <img src={"/cogWheel2.png"} alt="Cog Wheel"></img>
                </div>
            </div>
        </div>
    </>
  );
}

export default Header;
