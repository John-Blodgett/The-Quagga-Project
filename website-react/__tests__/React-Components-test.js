import React from 'react';
import renderer from 'react-test-renderer';
import Header from '../src/Header';
import Navbar from '../src/Navbar';

test('renders Header correctly', () => {
    const tree = renderer.create(<Header />).toJSON();
    expect(tree).toMatchSnapshot();
});

test('renders Navbar correctly', () => {
    const tree = renderer.create(<Navbar />).toJSON();
    expect(tree).toMatchSnapshot();
});

var sampleAssignment = {
    'name': 'HW1',
    'desc': 'HW1 is blah blah blah',
    'pointsPossible': 10,
    'grade' : 80,
    'className': 'CSC309'
    };
var sampleAssignment2 = {
    'name': 'HW2',
    'desc': 'HW2 is about doing stuff',
    'pointsPossible': 30,
    'grade' : 100,
    'className': 'CSC430'
    };
var sampleAssignment3 = {
  'name': 'HW3',
  'desc': 'HW is about 309',
  'pointsPossible': 30,
  'grade' : 50,
  'className': 'CSC309'
  };

var todoList = [sampleAssignment, sampleAssignment2, sampleAssignment3]
const CSC309Classes = [sampleAssignment3, sampleAssignment]
const CSC430Classes = [sampleAssignment2]

function classWithHighestGrade(todoList) {
  const sortedList = todoList.sort((ele1, ele2) => { 
    return ele2['grade'] - ele1['grade'];})
  return sortedList[0]
}
function classWithLowestGrade(todoList) {
  const sortedList = todoList.sort((ele1, ele2) => { 
    return ele1['grade'] - ele2['grade'];})
  return sortedList[0]
}

function singleClassAssignments(todoList, name) {
const singleClass = todoList.filter((ele1) => { 
  return ele1['className'] == name;})
return singleClass
}


test('Finds the highest grade in a given assignments list', () => {
  expect(classWithHighestGrade(todoList)).toStrictEqual(sampleAssignment2);
});

test('Finds the lowest grade in a given assignments list', () => {
  expect(classWithLowestGrade(todoList)).toStrictEqual(sampleAssignment3);
});

test('Filers for all 309 courses', () => {
  expect(singleClassAssignments(todoList, "CSC309")).toStrictEqual(CSC309Classes);
});

test('Filers for all 430 courses', () => {
  expect(singleClassAssignments(todoList, "CSC430")).toStrictEqual(CSC430Classes);
});