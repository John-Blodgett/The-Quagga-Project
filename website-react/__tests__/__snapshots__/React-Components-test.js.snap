// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Navbar correctly 1`] = `
<div
  id="web-navbar-master"
>
  <div
    id="navbar-component"
  >
    FAQ
  </div>
  <div
    id="navbar-component"
  >
    Personal To-do
  </div>
  <div
    id="navbar-component"
  >
    School To-do
  </div>
  <div
    id="navbar-component"
  >
    Completed
  </div>
  <div
    id="navbar-component"
  >
    About Quagga
  </div>
</div>
`;

exports[`renders correctly 1`] = `
<div
  id="web-header-master"
>
  <div
    id="zebra-background"
  >
    <div
      id="web-header-logo"
    >
      <img
        alt="Quagga Logo"
        src="/Logo.png"
      />
    </div>
    <div
      id="web-header-title"
    >
      Quagga
    </div>
    <div
      id="web-cog-wheel"
    >
      <img
        alt="Cog Wheel"
        src="/cogWheel2.png"
      />
    </div>
  </div>
</div>
`;
